### PR TITLE
 Use TempDirectory in TestEnvironmentProvider 

### PR DESCRIPTION
**Pull request by jnferner**
This adds a nice prefix to our generated directories

### CODE REVIEW

File: Test.Utility/TestEnvironmentProvider.cs [hunk 1]
```diff
@@ -53,21 +48,15 @@ public void Dispose()
                 File.SetAttributes(destinationPath, FileAttributes.Normal);
             }
 
-            Directory.Delete(RootDirectory, true);
+            _tempDirectory.Dispose();
         }
-
-        private static string GenerateParent()
+
```
Comment (bash) [inline Test.Utility/TestEnvironmentProvider.cs]: Remove whitespace on this line

File: Test.Utility/TestEnvironmentProvider.cs [hunk 2]
```diff
@@ -2,26 +2,29 @@
...
+        private readonly TempDirectory.TempDirectory _tempDirectory;
 
         public string RootDirectory { get; }
 
         public TestEnvironmentProvider(IReadOnlyCollection<TestFile> testFiles)
         {
             _testFiles = testFiles;
-
+            _tempDirectory = CreateTempDirectory();
+            RootDirectory = _tempDirectory.FullName;
```
Comment (bash) [inline Test.Utility/TestEnvironmentProvider.cs]: This could be a property with a getter fn.

File: Test.Utility/TestEnvironmentProvider.cs (hunk 1 above)
Comment (jnferner) [reply Test.Utility/TestEnvironmentProvider.cs]: https://github.com/messerli-informatik-ag/test-utility/pull/7/commits/f68766a81c35103374e1365b9d78dfae67a5344f

File: Test.Utility/TestEnvironmentProvider.cs (hunk 2 above)
Comment (jnferner) [reply Test.Utility/TestEnvironmentProvider.cs]: https://github.com/messerli-informatik-ag/test-utility/pull/7/commits/04aceaf2e3c7070c4eb11bb1e4dd2f498d3e5a36

**Pull request merged**